feat: add PARSE_DATE and FORMAT_DATE functions (#7733)
confluentinc/ksql: docs/developer-guide/ksqldb-reference/scalar-functions.md

"America/Los_Angeles", "PDT", "Europe/London". For more information on timestamp formats, see
[DateTimeFormatter](https://cnfl.io/java-dtf).

### `FORMAT_DATE`

```sql

> FORMAT_DATE(timestamp, 'yyyy-MM-dd')

Reviewer: Shouldn't this be `FORMAT_DATE(date, ...)`

```

> Converts DATE value into a string representing the date in the given format.

Reviewer 2: ```suggestion
Converts a DATE value into a string that represents the date in the given format.
```

> Single quotes in the timestamp format can be escaped with two successive single

Reviewer 2: ```suggestion
You can escape single-quote characters in the timestamp format by using two successive single
```

quotes, `''`, for example: `'yyyy-MM-dd''T'''`.

### `PARSE_DATE`

```sql
PARSE_DATE(col1, 'yyyy-MM-dd')
```

Converts a string representation of a date in the

> given format into a DATE value. Single quotes in the timestamp

Reviewer 2: ```suggestion
given format into a DATE value. You can escape single-quote characters in the timestamp
```

> format can be escaped with two successive single

Reviewer 2: as above
